consensus/src/execution_ctx.rs

```diff
@@ -5,180 +5,30 @@
-        ratification_handler: Arc<
-            Mutex<ratification::handler::RatificationHandler>,
-        >,
-    ) -> Self {
-        Self {
-            round,
-            join_set: JoinSet::new(),
-            iter,
-            proposal_handler,
-            validation_handler,
-            ratification_handler,
-            committees: Default::default(),
-        }
-    }
-
-    pub(crate) async fn collect_past_event(
-        &self,
-        ru: &RoundUpdate,
-        msg: &Message,
-    ) -> Option<Message> {
-        let committee = self.committees.get_committee(msg.header.get_step())?;
-        match msg.topic() {
-            node_data::message::Topics::Candidate => {
-                let mut handler = self.proposal_handler.lock().await;
-                _ = handler
-                    .collect_from_past(
-                        msg.clone(),
-                        ru,
-                        msg.header.iteration,
-                        committee,
-                    )
-                    .await;
-            }
-            node_data::message::Topics::Validation => {
-                let mut handler = self.validation_handler.lock().await;
-                if let Ok(Ready(m)) = handler
-                    .collect_from_past(
-                        msg.clone(),
-                        ru,
-                        msg.header.iteration,
-                        committee,
-                    )
-                    .await
-                {
-                    return Some(m);
-                }
-            }
-            node_data::message::Topics::Ratification => {
-                let mut handler = self.ratification_handler.lock().await;
-                if let Ok(Ready(m)) = handler
-                    .collect_from_past(
-                        msg.clone(),
-                        ru,
-                        msg.header.iteration,
-                        committee,
-                    )
-                    .await
-                {
-                    return Some(m);
-                }
-            }
-            _ => {}
-        };
-
-        None
-    }
-
-    pub(crate) fn on_begin(&mut self, iter: u8) {
-        self.iter = iter;
-    }
-
-    pub(crate) fn get_generator(&self, iter: u8) -> Option<PublicKeyBytes> {
-        let step = StepName::Proposal.to_step(iter);
-        self.committees
-            .get_committee(step)
-            .and_then(|c| c.iter().next().map(|p| *p.bytes()))
-    }
-
-    pub(crate) fn on_end(&mut self) {
-        debug!(
-            event = "iter completed",
-            len = self.join_set.len(),
-            round = self.round,
-            iter = self.iter,
-        );
-        self.join_set.abort_all();
-    }
-}
-
-impl<DB: Database> Drop for IterationCtx<DB> {
-    fn drop(&mut self) {
-        self.on_end();
-    }
-}
-
 /// ExecutionCtx encapsulates all data needed by a single step to be fully
```

Comment:
Maybe we should rename this to `StepCtx`?

Reply:
> /// ExecutionCtx encapsulates all data needed by a single step to be fully

The current description is outdated because  we might invoke `collect_from_past` function in multiple steps to handle messages from previous iterations. I'd suggest to revise the description while keeping the name.